addons/cargo/functions/fnc_handleDestroyed.sqf

```diff
@@ -19,14 +19,22 @@
 
 params ["_object"];
 
+private _killed = false;
+
+if (count _this > 1) then {_killed = true}; // The Killed event has 4 params. The Deleted event only has 1 param.
+
 private _loaded = _object getVariable [QGVAR(loaded), []];
 
 if (_loaded isNotEqualTo []) then {
     // Delete all cargo
     {
         if (_x isEqualType objNull) then {
-            detach _x;
-            deleteVehicle _x;
+            if (_killed && {random 100 < GVAR(unloadOnKilled)}) then {
+                [_x, _object] call ace_cargo_fnc_unloadItem;
```

Comment:
`unloadItem` modifies the loaded array, so you'll probably want `forEachReversed` or it will skip entries (I think, not tested)

also the func can fail if it can't find a unload pos
not sure exactly best way to handle but we might want to fallback to detach and delete?